docbrown/db/src/loaders.rs

```diff
@@ -152,25 +168,322 @@ pub mod csv {
+        let csv_path: PathBuf = [env!("CARGO_MANIFEST_DIR"), "resources/wrongheader/"]
+            .iter()
+            .collect();
+        let csv_loader = CsvLoader::new(Path::new(&csv_path));
+
+        let delimiter_string = ",";
+        let delimiter: u8 = delimiter_string.as_bytes()[0];
+
+        let has_header = true;
+        csv_loader
+            .set_header(has_header)
+            .set_delimiter(delimiter)
+            .load_into_graph(&g, |lotr: Lotr, g: &GraphDB| {
+                let src_id = calculate_hash(&lotr.src_id);
+                let dst_id = calculate_hash(&lotr.dst_id);
+                let time = lotr.time;
+
+                g.add_vertex(
+                    src_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_vertex(
+                    dst_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_edge(
+                    src_id,
+                    dst_id,
+                    time,
+                    &vec![(
+                        "name".to_string(),
+                        Prop::Str("Character Co-occurrence".to_string()),
+                    )],
+                );
+            })
+            .expect("Csv did not parse.");
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_wrong_delimiter() {
+        let g = GraphDB::new(2);
+        // todo: move file path to data module
+        let csv_path: PathBuf = [env!("CARGO_MANIFEST_DIR"), "resources/withheader"]
+            .iter()
+            .collect();
+        let csv_loader = CsvLoader::new(Path::new(&csv_path));
+        let delimiter_string = ".";
+        let delimiter: u8 = delimiter_string.as_bytes()[0];
+
+        let has_header = true;
+        csv_loader
+            .set_header(has_header)
+            .set_delimiter(delimiter)
+            .load_into_graph(&g, |lotr: Lotr, g: &GraphDB| {
+                let src_id = calculate_hash(&lotr.src_id);
+                let dst_id = calculate_hash(&lotr.dst_id);
+                let time = lotr.time;
+
+                g.add_vertex(
+                    src_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_vertex(
+                    dst_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_edge(
+                    src_id,
+                    dst_id,
+                    time,
+                    &vec![(
+                        "name".to_string(),
+                        Prop::Str("Character Co-occurrence".to_string()),
+                    )],
+                );
+            })
+            .expect("Csv did not parse.");
+    }
```

Comment:
Probably write a test with a file with a delimiter other than ','. For example, space?